c/bobyqa_c.f90

```diff
@@ -62,10 +66,22 @@ subroutine bobyqa_c(cobj_ptr, data_ptr, n, x, f, xl, xu, nf, rhobeg, rhoend, &
 
 ! Read the inputs and convert them to the Fortran side types
 x_loc = real(x, kind(x_loc))
-xl_loc = real(xl, kind(xl_loc))
-xu_loc = real(xu, kind(xu_loc))
-rhobeg_loc = real(rhobeg, kind(rhobeg_loc))
-rhoend_loc = real(rhoend, kind(rhoend_loc))
+if (C_ASSOCIATED(xl)) then
+    call C_F_POINTER(xl, xl_loc_interm, shape=[n])
+    call safealloc(xl_loc, int(n, IK))
+    xl_loc = real(xl_loc_interm, kind(xl_loc))
+end if
+if (C_ASSOCIATED(xu)) then
+    call C_F_POINTER(xu, xu_loc_interm, shape=[n])
+    call safealloc(xu_loc, int(n, IK))
+    xu_loc = real(xu_loc_interm, kind(xu_loc))
+end if
+if (.not. is_nan(rhobeg)) then
```

Comment:
I do not think it is necessary to  check whether `rhobeg` or `rhoend` is NaN, given 

https://github.com/libprima/prima/blob/main/fortran/common/preproc.f90#L296-L315

For any real scalar, the Fortran code will interpret NaN as an absent argument. Hence the C interface can simply pass whatever is received to the Fortran backend. 

Recall https://github.com/libprima/prima/pull/135#issuecomment-1897853547 ("After these changes, the C interface can simply pass whatever is received to Fortran.")

What do you think? 

Thank you.

Reply:
> I do not think it is necessary to check whether `rhobeg` or `rhoend` is NaN, given
> 
> https://github.com/libprima/prima/blob/f55179c0a4ef702eccd3e6f90637168e346ca044/fortran/common/preproc.f90#L312-L331
> 
> For any real scalar, the Fortran code will interpret NaN as an absent argument. Hence the C interface can simply pass whatever is received to the Fortran backend.
> 
> Recall [#135 (comment)](https://github.com/libprima/prima/pull/135#issuecomment-1897853547) ("After these changes, the C interface can simply pass whatever is received to Fortran.")
> 
> What do you think?
> 
> Thank you.

On my side, this is the only thing left before merging.